### PR TITLE
Added Support for Custom Labels and annotations on OpenMetadata Deployment Object

### LINKED ISSUE #69: Add Support for Custom Labels and annotations on OpenMetadata Deployment Object
Ref - https://openmetadata.slack.com/archives/C02B6955S4S/p1661982545124259

### PR DESCRIPTION
**Pull request by Ashok810**
### Describe your changes :

fixes #69 

#
### Type of change :

- [x] Improvement
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: charts/openmetadata/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -4,9 +4,10 @@ kind: ServiceAccount
 metadata:
   name: {{ include "OpenMetadata.serviceAccountName" . }}
   labels:
-    {{- include "OpenMetadata.labels" . | nindent 4 }}
-  {{- with .Values.serviceAccount.annotations }}
+    {{- include "OpenMetadata.labels" . | nindent 4 }} 
   annotations:
+    {{- include "OpenMetadata.annotations" . | nindent 4 }}
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/serviceaccount.yaml:9]: Surround this with if condition for `.Values.annotations` to be available.

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: charts/openmetadata/templates/deployment.yaml [hunk 2]
```diff
@@ -2,21 +2,24 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "OpenMetadata.fullname" . }}
+  annotations:
+    {{- include "OpenMetadata.annotations" . | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (Ashok810) [reply charts/openmetadata/templates/deployment.yaml]: Added If condition for .Values.annotations

File: charts/openmetadata/templates/deployment.yaml [hunk 3]
```diff
@@ -2,21 +2,24 @@ apiVersion: apps/v1
...
     matchLabels:
-      {{- include "OpenMetadata.selectorLabels" . | nindent 6 }}
+       {{- include "OpenMetadata.labels" . | nindent 6 }} 
   template:
     metadata:
       {{- with .Values.podAnnotations }}
       annotations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- include "OpenMetadata.annotations" . | nindent 8 }}
```
[the remark this answers is not in the document]
Comment (Ashok810) [reply charts/openmetadata/templates/deployment.yaml:23]: Added If condition for .Values.annotations

File: charts/openmetadata/templates/ingress.yaml [hunk 4]
```diff
@@ -17,11 +17,12 @@ kind: Ingress
 metadata:
   name: {{ $fullName }}
   labels:
-    {{- include "OpenMetadata.labels" . | nindent 4 }}
-  {{- with .Values.ingress.annotations }}
+    {{- include "OpenMetadata.labels" . | nindent 4 }} 
   annotations:
+    {{- include "OpenMetadata.annotations" . | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (Ashok810) [reply charts/openmetadata/templates/ingress.yaml:23]: Added If condition for .Values.annotations

File: charts/openmetadata/templates/service.yaml [hunk 5]
```diff
@@ -3,11 +3,12 @@ kind: Service
 metadata:
   name: {{ include "OpenMetadata.fullname" . }}
   labels:
-    {{- include "OpenMetadata.labels" . | nindent 4 }}
-  {{- with .Values.service.annotations }}
+    {{- include "OpenMetadata.labels" . | nindent 4 }} 
   annotations:
+    {{- include "OpenMetadata.annotations" . | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (Ashok810) [reply charts/openmetadata/templates/service.yaml:9]: Added If condition for .Values.annotations

**Pull request closed without merging**